src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest2.cs

```diff
@@ -165,6 +166,38 @@ public void NativeAOT ()
 				Assert.IsNotNull (method, $"{linkedMonoAndroidAssembly} should contain {typeName}.{methodName}");
 			}
 
+			var typemap = new Dictionary<string, TypeReference> ();
+			var linkedRuntimeAssembly = Path.Combine (intermediate, "linked", "Microsoft.Android.Runtime.NativeAOT.dll");
+			FileAssert.Exists (linkedRuntimeAssembly);
+			using (var assembly = AssemblyDefinition.ReadAssembly (linkedRuntimeAssembly)) {
+				var type = assembly.MainModule.Types.FirstOrDefault (t => t.Name == "NativeAotTypeManager");
+				Assert.IsNotNull (type, $"{linkedRuntimeAssembly} should contain NativeAotTypeManager");
+				var method = type.Methods.FirstOrDefault (m => m.Name == "InitializeTypeMappings");
+				Assert.IsNotNull (method, "NativeAotTypeManager should contain InitializeTypeMappings");
+
+				foreach (var i in method.Body.Instructions) {
+					if (i.OpCode != Mono.Cecil.Cil.OpCodes.Ldstr)
+						continue;
+					if (i.Operand is not string javaName)
+						continue;
+					if (i.Next.Operand is not TypeReference t)
+						continue;
+					typemap.Add (javaName, t);
+				}
+
+				// Basic types
+				AssertTypeMap ("java/lang/Object", "Java.Lang.Object");
+				AssertTypeMap ("java/lang/String", "Java.Lang.String");
+				AssertTypeMap ("android/app/Activity", "Android.App.Activity");
+				AssertTypeMap ("android/widget/Button", "Android.Widget.Button");
+
+				// Special *Invoker case
+				Assert.IsFalse (StringAssertEx.ContainsText (b.LastBuildOutput,
+					"ILLink: Duplicate typemap entry for android/view/View$OnClickListener => Android.Views.View/IOnClickListenerInvoker"),
+					"Should get log message about duplicate IOnClickListenerInvoker!");
+				AssertTypeMap ("android/view/View$OnClickListener", "Android.Views.View/IOnClickListener");
+			}
```

Comment:
Regarding prioritizing `Mono.Android` types over types from other assemblies, do we have a test case for it?

Reply:
This assertion is checking for now:
```csharp
AssertTypeMap ("java/lang/Object", "Java.Lang.Object");
```
Previously, it had `Java.Interop.JavaObject` from `Java.Interop.dll`.

In the future, we could probably check against a type created in a developer's assembly.